.taskcluster.yml

```diff
@@ -1,889 +1,77 @@
-        \ {{ event.head.repo.branch }} {{event.head.sha}} firefox-nightly &&\n   \
-        \         cd ~/web-platform-tests &&\n            ./tools/ci/ci_taskcluster.sh\
-        \ firefox testharness 15 15"]
-    image: gsnedders/web-platform-tests:0.13
-    maxRunTime: 7200
-  provisionerId: aws-provisioner-v1
-  workerType: wpt-docker-worker
-version: 0
+  $if: tasks_for == "github-push"
+  then:
+    $if: event.ref == "refs/heads/master"
+    then:
+      $flattenDeep:
+        $map: [{name: firefox, channel: nightly}, {name: chrome, channel: dev}]
+        each(browser):
+          $map:
+            - [testharness, 1, 15]
+            - [testharness, 2, 15]
+            - [testharness, 3, 15]
+            - [testharness, 4, 15]
+            - [testharness, 5, 15]
+            - [testharness, 6, 15]
+            - [testharness, 7, 15]
+            - [testharness, 8, 15]
+            - [testharness, 9, 15]
+            - [testharness, 10, 15]
+            - [testharness, 11, 15]
+            - [testharness, 12, 15]
+            - [testharness, 13, 15]
+            - [testharness, 14, 15]
+            - [testharness, 15, 15]
+            - [reftest, 1, 10]
+            - [reftest, 2, 10]
+            - [reftest, 3, 10]
+            - [reftest, 4, 10]
+            - [reftest, 5, 10]
+            - [reftest, 6, 10]
+            - [reftest, 7, 10]
+            - [reftest, 8, 10]
+            - [reftest, 9, 10]
+            - [reftest, 10, 10]
+            - [wdspec, 1, 1]
+          each(chunk):
+            taskId: {$eval: 'as_slugid(browser.name + browser.channel + chunk[0] + str(chunk[1]))'}
+            taskGroupId: {$eval: 'as_slugid("task group")'}
+            created: {$fromNow: ''}
+            deadline: {$fromNow: '2 hours'}
+            provisionerId: aws-provisioner-v1
+            # Contributors interested in configurating TaskCluster to run
+            # against their fork of WPT should change the `workerType` to
+            # "github-worker".
+            workerType: wpt-docker-worker
+            metadata:
+              name: wpt-${browser.name}-${browser.channel}-${chunk[0]}-${chunk[1]}
+              description: >-
+                A subset of WPT's "${chunk[0]}" tests (chunk number ${chunk[1]}
+                of ${chunk[2]}), run in the ${browser.channel} release of
+                ${browser.name}.
+              owner: ${event.pusher.email}
+              source: ${event.repository.url}
+            payload:
+              image: gsnedders/web-platform-tests:0.13
+              maxRunTime: 7200
+              artifacts:
+                public/results:
+                  path: /home/test/artifacts
+                  type: directory
+              command:
+                - /bin/bash
+                - --login
+                - -c
+                - "~/start.sh ${event.repository.url} ${event.ref[11:]} ${event.after} ${browser.name}-${browser.channel} &&
```

Comment:
> It's unclear why we're using event.ref[11:]. I assume that's stripping off
> refs/heads/ but if that's not actually needed then we should just use the
> full ref

`git clone`'s `--branch` option doesn't operate on fully-qualitifed refs like `refs/heads/master`.

    $ git clone git@github.com:web-platform-tests/wpt.git --branch refs/heads/master
    Cloning into 'wpt'...
    fatal: Remote branch refs/heads/master not found in upstream origin

>  otherwise comment on where 11 comes from.

Instead of a comment, maybe we can demonstrate the intention by calculating the constant value:

    ${event.ref[len('refs/heads/'):]}

What do you think?

> Also `event.after` doesn't seem to exist? I think you mean `event.head`.

I thought so too, at first, but [the event has no `head` property](https://github.com/bocoup/wpt/commit/8b01a0d4dfcf37271f3377f05319833a5894cf70#commitcomment-29770134)